### PR TITLE
Validate network when validating addresses

### PR DESCRIPTION
**Pull request by PawelPawelec-RDX**

### CODE REVIEW

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Migrations/20231130132946_AddNetworkHrpSuffixToNetworkConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.AddColumn<string>(
+                name: "network_hrp_suffix",
+                table: "network_configuration",
+                type: "text",
+                nullable: true,
+                defaultValue: null);
+
+            migrationBuilder.Sql("update network_configuration set network_hrp_suffix = (select substring(recordset.\"HrpPrefix\"::text,'_(.+)') from network_configuration, jsonb_to_recordset(address_type_definitions) as recordset(\"HrpPrefix\" TEXT) WHERE recordset.\"HrpPrefix\" like '%account%' limit 1)");
```
Comment (dhedey) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/Migrations/20231130132946_AddNetworkHrpSuffixToNetworkConfiguration.cs:84]: :D nice. I trust this has been tested 👍

**Review by dhedey — APPROVED** [review]
Thanks! Nice stuff. A couple of minor thoughts/suggestions - but nothing particularly important.

Approving subject to relevant minor tweaks / markups

File: src/RadixDlt.NetworkGateway.GatewayApi/Services/NetworkConfigurationProvider.cs [hunk 2]
```diff
@@ -126,6 +127,11 @@ public string GetNetworkName()
         return GetCapturedConfig().NetworkName;
     }
 
+    public string GetNetworkHrpSuffix()
+    {
+        return GetCapturedConfig().NetworkHrpSuffix;
```
[the remark this answers is not in the document]
Comment (dhedey) [reply src/RadixDlt.NetworkGateway.GatewayApi/Services/NetworkConfigurationProvider.cs:132]: Edit - sorry I missed the changes in `src/RadixDlt.NetworkGateway.PostgresIntegration/Services/CapturedConfigProvider.cs` all good!

File: src/RadixDlt.NetworkGateway.GatewayApi/Validators/RadixAddressValidator.cs [hunk 3]
```diff
@@ -63,59 +63,39 @@
...
 
-public sealed class RadixAddressValidator<T> : PropertyValidator<T, string?>
+public sealed class RadixAddressValidator : AbstractValidator<string>
 {
-    public override string Name => "RadixAddressValidator";
-
-    private readonly string? _expectedHrp;
-
-    public RadixAddressValidator(string? expectedHrp)
+    public RadixAddressValidator(INetworkAddressConfigProvider networkAddressConfigProvider)
```
[the remark this answers is not in the document]
Comment (PawelPawelec-RDX) [reply src/RadixDlt.NetworkGateway.GatewayApi/Validators/RadixAddressValidator.cs:75]: we weren't using it now, so i dropped that code. We can add it anytime.

**Pull request merged**